File: Introductory Problems/4. increasing array/increasing.js
function increasingArray(size, ...arr) {
  let moves = 0;
  for (let i = 0; i < size - 1; i++) {
    if (arr[i + 1] < arr[i]) {
      const diff = arr[i] - arr[i + 1];
      arr[i + 1] += diff;
      moves += diff;
    }
  }
  console.log(arr);
  return moves;
}

console.log(increasingArray(5, 3, 2, 5, 1, 7));

File: Introductory Problems/4. increasing array/increasing.py
def increasing_array(size, *arr):
    arr = list(arr)
    moves = 0
    for i, _ in enumerate(range(size-1)):
        if arr[i] > arr[i+1]:
            diff = arr[i] - arr[i+1]
            arr[i+1] += diff
            moves += diff
    print(arr)
    return moves

print(increasing_array(5, 3, 2, 5, 1, 7))
